storage/remote/client.go

```diff
@@ -198,7 +201,25 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, false, 0, 0}
+	}
+	if httpResp.StatusCode == 429 {
+		recErr := RecoverableError{err, true, 0, 0}
+		if after := httpResp.Header.Get("Retry-After"); after != "" {
+			duration, err := model.ParseDuration(after)
```

Comment:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.37 specifies that the `Retry-After` header can be either a duration in seconds, or a date time. If we support `Retry-After` we should support both of those forms, not the duration parsing that is done with `model.ParseDuration`.

Reply:
I have updated the logic to implement both the `seconds` way and the `date-time` one. I am not sure if the `date-time` one that I implemented is the best approach. So, a review would be helpful there as well.